src/mcl.jl

```diff
@@ -69,44 +69,23 @@ function _mcl_clusters(mcl_adj::Matrix{Float64}, allow_singles::Bool, zero_tol::
     clu_ixs, clu_sizes, unassigned_count
 end
 
-# adjacency matrix expansion (matrix-wise raising to a given integer power) kernel
-# FIXME `_mcl_expand!()` that does not allocate new expanded matrix
-function _mcl_expand(src::Matrix, expansion::Integer)
-    src ^ expansion
-end
-
-# adjacency matrix expansion (matrix-wise raising to a given power) kernel
-# FIXME `_mcl_expand!()` that does not allocate new expanded matrix
-function _mcl_expand(src::Matrix, expansion::Number)
-    # we have to implement the workarond for matrix-power because of julia bug #16930
-    isinteger(expansion) && return _mcl_expand(src, Integer(real(expansion)))
-    v, X = eig(src)
-    # FIXME type instability here, revisit when #16930 is fixed
-    (eltype(v) <: Complex) || (any(v.<0) && (v = complex(v)))
-    Xinv = ishermitian(src) ? X' : inv(X)
-    X * Diagonal(v.^expansion) * Xinv
+# adjacency matrix inflation (element-wise raising to a given power) kernel
+function _mcl_inflate!{T<:Real}(dest::Matrix{Float64}, src::Matrix{T}, inflation::Integer)
+    map!(el -> el^inflation, dest, src)
 end
 
 # adjacency matrix inflation (element-wise raising to a given power) kernel
-function _mcl_inflate!(dest::Matrix{Float64}, src::Matrix{Complex128}, inflation::Number)
-    src_norm = vecnorm(src)
-    min_rel = -1E-3*src_norm
-    min_img = 1E-3*src_norm
-    @inbounds for (i, el) in enumerate(src)
-        rel = real(el)
-        img = imag(el)
-        if rel < min_rel || (abs(img) > min_img && abs(img) > 1E-3*abs(rel))
-            throw(InexactError())
-        end
-        dest[i] = max(0.0, rel)^inflation
-    end
-    return dest
+function _mcl_inflate!{T<:Number}(dest::Matrix{Float64}, src::Matrix{T}, inflation::Number)
```

Comment:
I would add here a comment that it's non-integral power inflation branch and what issue `real(el+0im)` fixes.
`dest::Matrix{T}`?

Actually, what about `el_inflate(x::Number, inflation::Integer)` and `el_inflate(x::Number, inflation::Number)` to avoid the duplication of the loop code?

Reply:
However, as `inflation` is a keyarg parameter in `mcl()` and type inference of keyarg parameters would not be fixed until Julia v1.x, it's better not to call `el_inflate()` from `mcl()` directly.